tests/README.md

```diff
@@ -70,12 +88,59 @@ make tests args="[TESTFILE]::[TESTNAME]"
 where `TESTFILE` is as above, and `TESTNAME` is the name of the test
 within the test file (ex: `test_elf.py::test_virtual_valid`).
 
-See [pytest docs](https://docs.pytest.org/en/7.1.x/how-to/usage.html)
-for more invocation options.
+To run a single case of a single test:
+
+```sh
+# root of the repo
+make tests args="[TESTFILE]::[TESTNAME][test-case]"
+```
+
+ex:
+
+```sh
+make tests args="test_elf.py::test_virtual_valid[copy_files0]"
+```
+
+Any arguments passed in through `args` will be directly passed through to the underlying `pytest` call. See [pytest docs](https://docs.pytest.org/en/7.1.x/how-to/usage.html) for more invocation options.
+
+### Running Tests Directly
+
+While tests can be run directly via `pytest`, this is not recommended; they are intended to be run through the `docker compose` environment, and many of them are likely to fail without the correct setup.
+
+WARNING: These tests are ELF only.
+
+If you would really like to run tests directly, ensure that you have the following:
+
+- `python` version 3.11 or greater
+- `pipx` (recommended) or `pip` package installer for python
+- `docker` for the docker tests
+
+To set up the testing framework:
+
+1. Install poetry with `pipx install poetry`
+2. From the repository root: `cd ./tests`
+3. Install dependencies with `poetry install` (the list of dependencies to be installed is located at `tests/pyproject.toml`)
+
+To run the tests:
+
+1. From the repository root: `cd ./tests`
+2. Start poetry shell with `poetry shell`
+3. Run tests with `pytest` (flags and arguments as in the previous section)
```

Comment:
poetry shell has some implications which might break for some people. in docs it might be simpler to do `poetry run pytest ....` which internally will activate appropriate venv and then run pytest in that env without explicitly activating any of the shell bits (bash/zsh/fish/etc)